lib/facility.rb

```diff
@@ -1,14 +1,72 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, 
+              :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return unless @services.include?('Vehicle Registration')
+    return unless vehicle.is_a?(Vehicle)
+
+    update_plate_type_and_fees(vehicle)
+    set_registration_date(vehicle)
```

Comment:
I do like that you created helper methods to move some of the logic out of this register_vehicle method. Makes it a lot more readable! 
I do think we could talk more about which class these helper methods should be in though. The class that is the owner of the attribute we're changing should be the class that is changing the attribute. For example, `set_registration_date`, we're changing a Vehicle attribute, so we could put that method in the Vehicle class and then on line 23 call `vehicle.set_registration_date`.